hack/release

```diff
@@ -0,0 +1,65 @@
+#!/usr/bin/env bash
+
+set -eu -o pipefail
+
+: "${BUILDX_CMD=docker buildx}"
+: "${DESTDIR=./bin/release}"
+: "${CACHE_FROM=}"
+: "${CACHE_TO=}"
+
+: "${SIGN=}"
+: "${PFX=}"
+: "${PFXPASSWORD=}"
+
+if [ -n "$CACHE_FROM" ]; then
+  for cfrom in $CACHE_FROM; do
+    cacheFlags+=(--set "*.cache-from=$cfrom")
+  done
+fi
+if [ -n "$CACHE_TO" ]; then
+  for cto in $CACHE_TO; do
+    cacheFlags+=(--set "*.cache-to=$cto")
+  done
+fi
+
+dockerpfx=$(mktemp -t dockercredhelper-pfx.XXXXXXXXXX)
+function clean {
+  rm -f "$dockerpfx"
+}
+trap clean EXIT
+
+# release
+(
+  set -x
+  ${BUILDX_CMD} bake "${cacheFlags[@]}" --set "*.output=$DESTDIR" release
+)
+
+# wrap binaries
+mv -f ./${DESTDIR}/**/* ./${DESTDIR}/
+find ./${DESTDIR} -type d -empty -delete
+
+# sign binaries
+if [ -n "$SIGN" ]; then
+  for f in "${DESTDIR}"/*".darwin-"*; do
+    SIGNINGHASH=$(security find-identity -v -p codesigning | grep "Developer ID Application: Docker Inc" | cut -d ' ' -f 4)
+    xcrun -log codesign -s "$SIGNINGHASH" --force --verbose "$f"
+    xcrun codesign --verify --deep --strict --verbose=2 --display "$f"
+  done
+  for f in "${DESTDIR}"/*".windows-"*; do
+    echo ${PFX} | base64 -d > "$dockerpfx"
+    signtool sign /fd SHA256 /a /f pfx /p ${PFXPASSWORD} /d Docker /du https://www.docker.com /t http://timestamp.verisign.com/scripts/timestamp.dll "$f"
+  done
+fi
+
+# checksums
+if ! command shasum --version > /dev/null 2>&1; then
+  echo >&2 "ERROR: shasum is required"
+  exit 1
+fi
+find ./${DESTDIR}/ -type f \( -iname "docker-credential-*" \) -print0 | sort -z | xargs -r0 shasum -a 256 -b | sed 's# .*/#  #' > ./${DESTDIR}/checksums.txt
+
+# verify
+(
+  cd ./${DESTDIR}
+  shasum -a 256 -U -c checksums.txt
```

Comment:
Looks like macOS / BSD and GNU are different here as well; the -U option is named -p on macOS;

On Linux:

```
 -U, --UNIVERSAL   read in Universal Newlines mode
                        produces same digest on Windows/Unix/Mac
```

On macOS

```
  -p, --portable    read in portable mode
                        produces same digest on Windows/Unix/Mac
```

So I guess we need to hack around that;

```suggestion
	if [ "$(uname -s)" = "Darwin" ]; then
		shasum -a 256 -p -c checksums.txt
	else
		shasum -a 256 -U -c checksums.txt
	fi
```

Reply:
> On macOS
> 
> ```
>   -p, --portable    read in portable mode
>                         produces same digest on Windows/Unix/Mac
> ```

I think you're using a pretty old version of perl utils. What's the output of `shasum -v`?

On my side on M1 (21.6.0) I don't have the `-p` flag available:

```console
$ uname -v
Darwin Kernel Version 21.6.0: Sat Jun 18 17:05:47 PDT 2022; root:xnu-8020.140.41~1/RELEASE_ARM64_T8101
$ shasum --help
Usage: shasum [OPTION]... [FILE]...
Print or check SHA checksums.
With no FILE, or when FILE is -, read standard input.

  -a, --algorithm   1 (default), 224, 256, 384, 512, 512224, 512256
  -b, --binary      read in binary mode
  -c, --check       read SHA sums from the FILEs and check them
      --tag         create a BSD-style checksum
  -t, --text        read in text mode (default)
  -U, --UNIVERSAL   read in Universal Newlines mode
                        produces same digest on Windows/Unix/Mac
  -0, --01          read in BITS mode
                        ASCII '0' interpreted as 0-bit,
                        ASCII '1' interpreted as 1-bit,
                        all other characters ignored

The following five options are useful only when verifying checksums:
      --ignore-missing  don't fail or report status for missing files
  -q, --quiet           don't print OK for each successfully verified file
  -s, --status          don't output anything, status code shows success
      --strict          exit non-zero for improperly formatted checksum lines
  -w, --warn            warn about improperly formatted checksum lines

  -h, --help        display this help and exit
  -v, --version     output version information and exit

When verifying SHA-512/224 or SHA-512/256 checksums, indicate the
algorithm explicitly using the -a option, e.g.

  shasum -a 512224 -c checksumfile

The sums are computed as described in FIPS PUB 180-4.  When checking,
the input should be a former output of this program.  The default
mode is to print a line with checksum, a character indicating type
(`*' for binary, ` ' for text, `U' for UNIVERSAL, `^' for BITS),
and name for each FILE.  The line starts with a `\' character if the
FILE name contains either newlines or backslashes, which are then
replaced by the two-character sequences `\n' and `\\' respectively.

Report shasum bugs to mshelor@cpan.org
```